### ISSUE TITLE
"or" when evaluating network 

### ISSUE THREAD

**Issue by BenCoke12** [vehicle-lang/vehicle#641]
When evaluating a network the property:

```    
let scores = classifier (images ! 0) in
    scores ! 7 > scores ! 2 or not(scores ! 7 > scores ! 2)
```

produces the message "proved no witness exists" on all the images I have tested. This property should produce the message "witness found" for all inputs as the statement: A or not(A) is true for all values of A.
The code is:

```
numberOfClasses = 10
type Class = Index numberOfClasses

type Image = Tensor Rat [28, 28]

type Score = Rat

@network
classifier : Image -> Vector Score numberOfClasses

@dataset
images : Vector Image 1

@property
pulloverLowScore : Bool
pulloverLowScore = 
    let scores = classifier (images ! 0) in
    scores ! 7 > scores ! 2 or not(scores ! 7 > scores ! 2)
```

**Comment by BenCoke12** [comment]
Network is here: https://github.com/BenCoke12/FashionVerification/blob/main/onnxnetworks/fashion1l32n.onnx
Data is this: https://github.com/BenCoke12/FashionVerification/blob/main/idxdata/individuals/Image0.idx
Specification is this: https://github.com/BenCoke12/FashionVerification/blob/main/vclspecs/debuggingOr.vcl

The command I used is: vehicle verify --specification vclspecs/debuggingOr.vcl --network classifier:onnxnetworks/fashion1l32n.onnx --dataset images:idxdata/individuals/Image1.idx --verifier Marabou --verifierLocation ../Marabou/build/Marabou

The expected result is: result: 🗸 - witness found
The actual result is: result: ✗ - proved no witness exists

**Comment by MatthewDaggitt** [comment]
Further minimised example:
```
type Image = Tensor Rat [28, 28]

@network
classifier : Image -> Vector Rat 10

@dataset
images : Vector Image 1

@property
pulloverLowScore : Bool
pulloverLowScore = 
    let scores = classifier (images ! 0) in
    scores ! 7 > 0 or not(scores ! 7 > 0)
```

**Comment by MatthewDaggitt** [comment]
Thanks for the detailed bug report! Fixed in #648
